src/fields/field2_128/extend_constants.rs

```diff
@@ -0,0 +1,404 @@
+
+/// The novel polynomial basis used to inject integers into [`Field2_128`] so that they can be
+/// efficiently multiplied with other field elements.
+///
+/// The generator is g=x^{(2^{128}-1) / (2^{16}-1)} and the basis consists of g^i for 0 <= i <
+/// 16. Despite the actual field having 128 bits, this 16 bit basis is large enough for values
+/// in the Longfellow commitment scheme.
+///
+/// Described in section 3.3 of [Longfellow][1], and in [section 2.2.2][2] of the specification.
+///
+/// [1]: https://eprint.iacr.org/2024/2010.pdf
+/// [2]: https://datatracker.ietf.org/doc/html/draft-google-cfrg-libzk-01#section-2.2.2
+// Restrict this to test cfg for now, to avoid a dead_code warning
+pub(crate) const fn subfield_basis() -> [Field2_128; Field2_128::SUBFIELD_BIT_LENGTH] {
+    // Computed in SageMath:
+    // GF2 = GF(2)
+    // x = polygen(GF2)
+    // GF2_128.<x> = GF2.extension(x^128 + x^7 + x^2 + x + 1)
+    // # (2^128 - 1) / (2^16 - 1) = 5192376087906286159508272029171713
+    // g = GF2_128(x)^5192376087906286159508272029171713
+    // for i in range(16):
+    //     print((g^i).to_integer())
+    [
+        Field2_128::from_u128_const(1),
+        Field2_128::from_u128_const(122753392676920971658749122761936853580),
+        Field2_128::from_u128_const(185726315739164108859399796142663757480),
+        Field2_128::from_u128_const(242075443675499227109267320400911498989),
+        Field2_128::from_u128_const(222542784214874944227167260511584909351),
+        Field2_128::from_u128_const(112425233274138136655968706803108112542),
+        Field2_128::from_u128_const(111955919296675432803292387102199683976),
+        Field2_128::from_u128_const(98263563783606923211003062286538201250),
+        Field2_128::from_u128_const(201564998845547644025367330818532893392),
+        Field2_128::from_u128_const(160337305018218142404182087593505552404),
+        Field2_128::from_u128_const(226322062164902409880343938983895220315),
+        Field2_128::from_u128_const(162097237762330204766181745825071946233),
+        Field2_128::from_u128_const(231881152083561309652087018093156258488),
+        Field2_128::from_u128_const(285712114744156533702206238221900352048),
+        Field2_128::from_u128_const(7298129229485713500594372746195714592),
+        Field2_128::from_u128_const(33006046103584326006971076539670943571),
+    ]
+}
+
+/// The precomputed array used in the twiddle function for additive Fast Fourier Transforms in
+/// [`Field2_128`]. Also "W^hat" in section 3.2 of the [paper][1], or the normalized subspace vanishing polynomials.
+///
+/// Element [i][j] of this array represents W^hat_i(beta_j).
+///
+/// [1]: https://eprint.iacr.org/2024/2010.pdf
+pub(crate) const fn twiddle_array()
+-> [[Field2_128; Field2_128::SUBFIELD_BIT_LENGTH]; Field2_128::SUBFIELD_BIT_LENGTH] {
+    // Computed using fields::field2_128::extend_constants::test::compute_twiddle_array. We can't
+    // evaluate that function in const because it uses loops.
```

Comment:
nit: Loops themselves are not the issue preventing us from evaluating this in a const context, but idiomatic loops don't work due to their reliance on traits.
```suggestion
    // evaluate that function in const because it uses lots of trait method calls.
```